### PR TITLE
Utf-8 csv files for UN/LOCODE file

### PR DESCRIPTION
**Pull request by End-rey**
Refs: #25.
I tried different encodings on the files and none worked. I also looked at other file sources and found a solution that if you pull files from the mdb extension, they are saved in utf8 and with the correct characters.

### CODE REVIEW

**Comment by roman-khimov** [comment]
Or maybe it's just CP1252, at least as suggested by https://github.com/datasets/un-locode/blob/main/scripts/prepare.py and nearby code.

**Comment by End-rey** [comment]
> Have you checked https://github.com/datasets/un-locode?

Yes, when I was solving the problem, I peeked into this repository for script.


> Or maybe it's just CP1252, at least as suggested by https://github.com/datasets/un-locode/blob/main/scripts/prepare.py and nearby code.

I tried cp1252 with files, that i get from https://service.unece.org/trade/locode/loc241csv.zip, and it not worked. 
I tried running the python code "prepare.py" and the same encoding problem occurs there. But if you have data with correct characters and run "integrate.py", it seems that pandas will update the data with the merge function and will not break the encoding.

I was only able to get files with the correct characters from the mdb file, so I'll try to search Go package to deal with mdb

**Comment by roman-khimov** [comment]
> I peeked into this repository for script.

Does it have correct UTF-8 in their DBs? Because we can fetch them instead of UN DBs if they're fine. This just needs to be verified.

**Comment by End-rey** [comment]
> Does it have correct UTF-8 in their DBs?

Yes, because they just get files from unece.org and process them so they don't have encoding problems.

How we can fetch there? Just wget the raw file from github?

**Comment by roman-khimov** [comment]
We can add it as git submodule or fetch raw files from GH. Both ways work for me.

File: internal/parsers/table/csv/calls.go [hunk 1]
```diff
@@ -49,6 +49,8 @@ const (
 	subDivCountry
 	subDivSubdivision
 	subDivName
+	_
```
Comment (AliceInHunterland) [inline internal/parsers/table/csv/calls.go:52]: You have deleted this line in the previous commit. So no need to add them back

File: unlocode-mdb-to-csv.sh [hunk 2]
```diff
@@ -1,28 +0,0 @@
-#!/bin/bash
```
Comment (AliceInHunterland) [inline unlocode-mdb-to-csv.sh]: Probably commits in this PR should be rebased into 2 commits. One about new source of data and another one about version changes.

Comment (AliceInHunterland) [reply unlocode-mdb-to-csv.sh]: Also, add notes about it to the change log.

Comment (End-rey) [reply unlocode-mdb-to-csv.sh]: As planned, upgrade to the new version will be in a separate pr

**Review by carpawell — COMMENTED** [review]
That is not exactly "git" as it is said in the commit, `git` != GitHub. Commits usually start with an imperative verb (e.g. "Get unlocode files from..."). "Ref" can be changed to "Closes" and merging will automatically close the issue.
Otherwise, I am OK.

File: Makefile [hunk 3]
```diff
@@ -3,7 +3,7 @@
 VERSION ?= "$(shell git describe --tags --match "v*" --dirty --always --abbrev=8 2>/dev/null || cat VERSION 2>/dev/null || echo "develop")"
 LOCODECLI ?= locode-db
 LOCODEDB ?= pkg/locodedb/data
-UNLOCODEFILE = loc232csv.zip
+UNLOCODEREVIOSION = 1b2c902d2f0057b36d800d8f25f40234412858cf
 UNLOCODEPREFIX = "2023-2"
```
[the remark this answers is not in the document]
Comment (End-rey) [reply Makefile]: Yes, I deleted it.

File: Makefile [hunk 4]
```diff
@@ -27,11 +27,12 @@ in/countries.dat:
 in/continents.geojson: continents.geojson.gz
 	gunzip -c $< > in/continents.geojson
 
-tmp/$(UNLOCODEFILE): | tmp
-	wget -c https://service.unece.org/trade/locode/$(UNLOCODEFILE) -O $@
-
-unlocode: tmp/$(UNLOCODEFILE) | in
-	unzip -u tmp/$(UNLOCODEFILE) -d in/
+unlocode:  | in
```
[the remark this answers is not in the document]
Comment (End-rey) [reply Makefile]: Delete space.

File: Makefile [hunk 5]
```diff
@@ -3,8 +3,7 @@
 VERSION ?= "$(shell git describe --tags --match "v*" --dirty --always --abbrev=8 2>/dev/null || cat VERSION 2>/dev/null || echo "develop")"
 LOCODECLI ?= locode-db
 LOCODEDB ?= pkg/locodedb/data
-UNLOCODEFILE = loc232csv.zip
-UNLOCODEPREFIX = "2023-2"
+UNLOCODEREVIOSION = 1b2c902d2f0057b36d800d8f25f40234412858cf
```
Comment (roman-khimov) [inline Makefile:6]: `UNLOCODEREVISION`

Comment (roman-khimov) [reply Makefile:6]: Why not 3648bfa776701c329d27136bef29fb3e21853f20, btw?

File: Makefile [hunk 6]
```diff
@@ -27,11 +26,12 @@ in/countries.dat:
...
-	wget -c https://service.unece.org/trade/locode/$(UNLOCODEFILE) -O $@
-
-unlocode: tmp/$(UNLOCODEFILE) | in
-	unzip -u tmp/$(UNLOCODEFILE) -d in/
+unlocode: | in
+	wget -c https://raw.githubusercontent.com/datasets/un-locode/${UNLOCODEREVIOSION}/data/subdivision-codes.csv -O in/SubdivisionCodes.tmp.csv
+	wget -c https://raw.githubusercontent.com/datasets/un-locode/${UNLOCODEREVIOSION}/data/code-list.csv -O in/CodeList.tmp.csv
+	awk 'NR > 1' in/SubdivisionCodes.tmp.csv > in/SubdivisionCodes.csv
+	awk 'NR > 1' in/CodeList.tmp.csv > in/CodeList.csv
+	rm in/*.tmp.csv
```
Comment (roman-khimov) [inline Makefile:34]: This is not a proper `Makefile`, it's a script (see deleted lines for how it can work in a better way with dependencies).

**Review by roman-khimov — COMMENTED** [review]
So, the DB is the same after this? I expected some updates to `pkg/locodedb/data`.

File: Makefile (hunk 6 above)
Comment (End-rey) [reply Makefile]: Split into files targets. Is that what you needed?

Comment (roman-khimov) [reply Makefile]: Much better this way. I'd still keep downloaded files, but this works too.

**Comment by roman-khimov** [comment]
Now please reorganize commits (fixes to previous commits are not expected in general) and we're good to go.

**Comment by roman-khimov** [comment]
CSVs are easy to diff, so it's important to take a look at what are the changes.
```
 pr-37 |20805 +++++++++++++++++++++++++++++++++---------------------------------
 1 file changed, 10428 insertions(+), 10377 deletions(-)
```
A lot of changes like
```
-AEAMF,Musaffah,5,AZ,Abu Z¸aby [Abu Dhabi],24.383333,54.483334
+AEAMF,Musaffah,5,AZ,Abū Z̧aby [Abu Dhabi],24.383333,54.483334
...
-AERWP,Ruwais Port,5,AZ,Abu Z¸aby [Abu Dhabi],24.1,52.716667
-AESHJ,Sharjah,5,SH,Ash Shariqah [Sharjah],25.35,55.383335
+AERWP,Ruwais Port,5,AZ,Abū Z̧aby [Abu Dhabi],24.1,52.716667
+AESHJ,Sharjah,5,SH,Ash Shāriqah [Sharjah],25.35,55.383335
...
-AOMTL,Matala,2,HUI,Huيla,-14.733334,15.033334
+AOMTL,Matala,2,HUI,Huíla,-14.733334,15.033334
```
as expected.

Also there are fixes for cases like
```
SISTS,Sveta Trojica v Slovenskih goricah,1,204,"Sveta Trojica v Slovenskih
goricah",46.566666,15.866667
```
Which are valid CSVs, but still not something you'd expect. And new ones are being added like
```
+VNVUT,Vung Tau,5,43,Bà Rịa - Vũng Tàu,10.35,107.066666
```
Seems to be OK overall.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**